core/jobs/batch_jobs/exp_validation_jobs.py

```diff
@@ -0,0 +1,828 @@
+                        if (rte_component['customization_args']
+                        ['text-with-value']) == '':
+                            rte_components_errors.append(
+                                'State - ' + str(key) +
+                                ' Link tag text value' + ' is empty.')
+
+                        if url_value != 'https':
+                            rte_components_errors.append(
+                                'State - ' + str(key) +
+                                ' Link tag url value' + ' does not start '
+                                + 'with https.')
+
+            states_with_values.append(
+                {'state_name': key,
+                'rte_components_errors': rte_components_errors}
+            )
+        return states_with_values
+
+    def filter_invalid_state_interaction_values_part_1(
+        self, states_dict: dict[str, state_domain.State]) -> list[dict]:
+        """Returns the errored state interaction values for
+            - FractionInput
+            - NumericInput
+            - NumberWithUnits
+
+        Args:
+            states_dict: dict[str, State]. The dictionary containing
+                state name as key and State object as value.
+
+        Returns:
+            states_with_values: list[dict]. The list of dictionaries
+            containing the errored values.
+        """
+        states_with_values = []
+
+        for key, value in states_dict.items():
+            number_with_units_rules = []
+            numeric_input_interaction_values = []
+            fraction_interaction_invalid_values = []
+            number_with_units_errors = []
+
+            answer_groups = value.interaction.answer_groups
+            for answer_group in answer_groups:
+                if value.interaction.id == 'FractionInput':
+                    inputs_with_whole_nums = ['HasDenominatorEqualTo',
+                    'HasNumeratorEqualTo', 'HasIntegerPartEqualTo',
+                    'HasNoFractionalPart']
+                    for rule_spec in answer_group.rule_specs:
+                        if rule_spec.rule_type not in inputs_with_whole_nums:
+                            num = rule_spec.inputs['f']['numerator']
+                            den = rule_spec.inputs['f']['denominator']
+                            whole = rule_spec.inputs['f']['wholeNumber']
+
+                            if den == 0:
+                                fraction_interaction_invalid_values.append(
+                                    'The rule ' +
+                                    str(answer_group.rule_specs.index(
+                                    rule_spec)) + ' of answer group ' +
+                                    str(answer_groups.index(answer_group)) +
+                                    ' has denominator equals to zero.')
+
+                            if (value.interaction.customization_args
+                            ['requireSimplestForm'].value) is True:
```

Comment:
Try to keep the ifs on one line, if possible. If you need more space, feel free to separate the code in a new function.

Reply:
Done